File: Python/8py 1/enough.py
"""
|--------------------------------------------------------------------------
| Will there be enough space?
|--------------------------------------------------------------------------
|
"""

def enough(cap, on, wait):
   return max(wait - (cap - on), 0)


def enough(cap, on, wait):
   return wait + on - cap if wait + on > cap else 0


def enough(cap, on, wait):
   return max(wait + on - cap, 0)


print(enough(10, 5, 5), 0)
print(enough(100, 60, 50), 10)
print(enough(20, 5, 5), 0)

File: JavaScript/8js 1/enough.js
/*
|--------------------------------------------------------------------------
| Will there be enough space?
|--------------------------------------------------------------------------
|
*/

function enough(cap, on, wait) {
   return wait - (cap - on) > 0 ? wait - (cap - on) : 0
}


function enough(cap, on, wait) {
   return Math.max(wait + on - cap, 0);
}


function enough(cap, on, wait) {
   return Math.abs(Math.min([cap - on - wait], 0))
}


console.log(enough(10, 5, 5), 0);
console.log(enough(100, 60, 50), 10);
console.log(enough(20, 5, 5), 0);